.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: Run CI
+on:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md' # Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  linux-build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version: [8.1, 8.2]
+        use-lowest-dependencies: [true, false]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # If you only need the current version keep this.
```

Comment:
This comment has stayed with us forever, hasn't it. `fetch-depth: 0` means fetch _all_ history. What a confusing UX choice, though I get why they made it.